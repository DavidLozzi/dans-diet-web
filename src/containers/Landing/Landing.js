import React, { useEffect, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Grid, makeStyles, Typography } from '@material-ui/core';
import AddCircleOutlineOutlinedIcon from '@material-ui/icons/AddCircleOutlineOutlined';

import { actions as myDietActions, name as dietName } from 'redux/api/myDiet/myDiet';
import Layout from 'containers/Layout/Layout';
import DietCard from 'components/DietCard/DietCard';
import ManageDietDialog from 'components/ManageDietDialog/ManageDietDialog';

import CONFIG from 'config';

const useStyles = makeStyles((theme) => ({
  root: {
    width: '100%',
    marginTop: theme.spacing(2)
  },
  gridList: {
    flexWrap: 'nowrap',
    // Promote the list into his own layer on Chrome. This cost memory but helps keeping high FPS.
    transform: 'translateZ(0)',
  },
  noDiets: {
    margin: theme.spacing(5)
  }
}));


const Landing = () => {
  const dispatch = useDispatch();
  const myDiets = useSelector((state) => state[dietName].diets);
  const loading = useSelector((state) => state[dietName].loading);
  const [openDietDetails, setOpenDietDetails] = useState(false);
  const [dietDetailsAction, setDietDetailsActions] = useState(CONFIG.OPTIONS.ADD);
  const [selectedDiet, setSelectedDiet] = useState();
  const classes = useStyles();

  useEffect(() => {
    myDietActions.loadDiets()(dispatch);
  }, [dispatch]);

  const toggleDietDetails = () => {
    setOpenDietDetails(!openDietDetails);
    if (openDietDetails) setSelectedDiet();
  };

  const manageDietDetail = (diet) => {
    setSelectedDiet(diet);
    setDietDetailsActions(CONFIG.OPTIONS.EDIT);
    setOpenDietDetails(true);
  };

  return (
    <Layout showTopNav showBottomNav>
      <Grid
        container
        className={classes.root}
      >
        <Grid item xs={1} />
        <Grid item xs={10}>
          <Typography
            variant="h3"
            gutterBottom
          >
            My Diets
          </Typography>
        </Grid>
        <Grid item xs={1} style={{ textAlign: 'right', alignContent: 'right' }}>
          <AddCircleOutlineOutlinedIcon fontSize="large" onClick={toggleDietDetails} />
          <ManageDietDialog
            openDietDetails={openDietDetails}
            toggleDietDetails={toggleDietDetails}
            diet={selectedDiet}
            dietDetailsAction={dietDetailsAction}
          />
        </Grid>
        {myDiets.map((diet) => (
          <Grid item key={diet.id} xs={12} sm={6}>
            <DietCard diet={diet} onEdit={manageDietDetail} />
          </Grid>
        ))}
        {
          !loading && myDiets.length === 0 &&
          <Typography variant="h5" className={classes.noDiets}>No diets yet, click the plus above to add your first one!</Typography>
        }
        {
          loading &&
          <div>loading</div>
        }
      </Grid>
    </Layout>
  );
};

export default Landing;