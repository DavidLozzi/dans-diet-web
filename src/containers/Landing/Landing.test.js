import React from 'react';
import { shallow } from 'enzyme';
import Landing from './Landing';

afterEach(() => {
  jest.resetAllMocks();
});

let component;
beforeAll(() => {
  component = shallow(<Landing />);
});

describe('Landing', () => {
  it('renders and matches snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});