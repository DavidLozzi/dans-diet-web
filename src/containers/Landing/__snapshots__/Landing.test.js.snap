// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing renders and matches snapshot 1`] = `ShallowWrapper {}`;
